### PR TITLE
feat: check ctype hash and trusted attester list

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes NO TICKET

Adds the latest SKYC Email CType.

This checks the attester of the credential against the list of ctypes and the corresponding ctype.

## How to test:
Please provide a brief step-by-step instruction.
If necessary provide information about dependencies (specific configuration, branches, database dumps, etc.)

- Step 1
- Step 2
- etc.

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: backend/src/credentials/verifySubmittedCredential.ts [hunk 1]
```diff
@@ -81,13 +81,22 @@ export async function verifySubmittedCredential(
...
+    if (ctype.cTypeHash !== credential.claim.cTypeHash) {
+      throw new Error(
+        'CType hash does not match with the corresponding trusted attester'
+      )
+    }
+    return ctype.trustedAttesters[0] === attesterOfTheirCredential
+  })
 
   // If you don't include a list of trusted attester on the credential-request, this check would be skipped
   if (ourTrustedAttesters) {
```
Comment (Dudleyneedham) [inline backend/src/credentials/verifySubmittedCredential.ts:98]: I think I could kill this line.

File: backend/src/credentials/verifySubmittedCredential.ts [hunk 2]
```diff
@@ -81,13 +81,22 @@ export async function verifySubmittedCredential(
...
+    if (!ctype.trustedAttesters) {
+      throw new Error('no trusted attester')
+    }
+    if (ctype.cTypeHash !== credential.claim.cTypeHash) {
+      throw new Error(
+        'CType hash does not match with the corresponding trusted attester'
+      )
+    }
+    return ctype.trustedAttesters[0] === attesterOfTheirCredential
+  })
```
Comment (kilted-andres) [inline backend/src/credentials/verifySubmittedCredential.ts:95]:
```suggestion
    const ourTrustedAttesters = cTypeRequested.cTypes.find((ctype) => {
    ctype.cTypeHash !== credential.claim.cTypeHash
  })?.trustedAttesters
```

File: backend/src/credentials/verifySubmittedCredential.ts [hunk 3]
```diff
@@ -81,13 +81,22 @@ export async function verifySubmittedCredential(
...
 
   // If you don't include a list of trusted attester on the credential-request, this check would be skipped
   if (ourTrustedAttesters) {
-    if (!ourTrustedAttesters.includes(attesterOfTheirCredential)) {
+    if (!ourTrustedAttesters) {
```
Comment (kilted-andres) [inline backend/src/credentials/verifySubmittedCredential.ts:99]:
```suggestion
  if (ourTrustedAttesters?.includes(attesterOfTheirCredential)) {
```

File: backend/src/credentials/listOfRequests.ts [hunk 4]
```diff
@@ -14,6 +14,14 @@ export const emailRequest: Kilt.IRequestCredentialContent = {
...
       ],
       requiredProperties: ['Email']
+    },
+    {
+      cTypeHash:
+        '0xae5bc64e500eb576b7b137288cec5d532094e103be46872f1ad54641e477d9fe',
+      trustedAttesters: [
+        'did:kilt:4pehddkhEanexVTTzWAtrrfo2R7xPnePpuiJLC7shQU894aY'
+      ],
+      requiredProperties: ['Email']
```
Comment (kilted-andres) [inline backend/src/credentials/listOfRequests.ts:24]: nice

**Review by kilted-andres — CHANGES_REQUESTED** [review]

**Review by kilted-andres — APPROVED** [review]
Great!

**Pull request merged**